app/assets/javascripts/icecomm.js

```diff
@@ -1,207 +1,770 @@
+  return browserType === 'Firefox';
+}
+
+module.exports = browserHandler;
+},{"./adapter":2}],4:[function(require,module,exports){
+var utils = require('./utils');
+var s = require('./streamHandler');
+var EventEmitter = require('./EventEmitter');
+
+var dataChannelHandler = {};
+var dataChannels = {};
+var pendingData = {};
+
+dataChannelHandler.addListenersToDataChannel = addListenersToDataChannel;
+dataChannelHandler.createDataChannel = createDataChannel;
+dataChannelHandler.setChannel = setChannel;
+dataChannelHandler.send = send;
+dataChannelHandler.removeDataChannel = removeDataChannel;
+dataChannelHandler.isOpen = isOpen;
+dataChannelHandler.isPending = isPending;
+dataChannelHandler.setPending = setPending;
+
+function removeDataChannel(callerID) {
+  if (dataChannels[callerID]) {
+    dataChannels[callerID] = undefined;
+  }
+}
+
+function addListenersToDataChannel(callerID) {
+
+  dataChannels[callerID].channel.onerror = utils.errorHandler;
+
+  dataChannels[callerID].channel.onmessage = function(event) {
+    messageDataHandler(callerID, event.data)
+  }
+
+  dataChannels[callerID].channel.onopen = function(event) {
+    openDataHandler(callerID);
+  }
+
+  dataChannels[callerID].channel.onclose = closeDataHandler
+}
+
+function createDataChannel(callerID, localPeerConnections) {
+  dataChannels[callerID] = {};
+  dataChannels[callerID].status = 'pending';
+  dataChannels[callerID].channel = localPeerConnections[callerID].channel.createDataChannel('RTCDataChannel', {reliable: true});
+}
+
+function setChannel(callerID, channel) {
+  dataChannels[callerID] = {};
+  dataChannels[callerID].status = 'pending';
+  dataChannels[callerID].channel = channel;
+}
+
+function send(callerID, data) {
+  if (data) {
+    dataChannels[callerID].channel.send(data);
+  }
+}
+
+function messageDataHandler(callerID, data) {
+  var options = {};
+  options.data = JSON.parse(data);
+  options.callerID = callerID;
+  EventEmitter.trigger('data', options);
+}
+
+function openDataHandler(callerID) {
+  utils.printDebugMessage('data channels open');
+  sendIfPending(callerID);
+  setOpen(callerID);
+  var options = s.createStreamOptions('remote', callerID);
+  EventEmitter.trigger('connected', options);
+}
+
+function closeDataHandler() {
+  utils.printDebugMessage('ON CLOSE CALLED');
+}
+
+function onLeftHandler(callerID) {
+  pc.removePeer(callerID);
+  var options = {};
+  options.callerID = callerID;
+  EventEmitter.trigger('disconnect', options);
+}
+
+function sendIfPending(callerID) {
+  if (dataChannels[callerID].status === 'pending') {
+    utils.printDebugMessage('sending pending data');
+    var data = pendingData[callerID];
+    send(callerID, data);
+    delete pendingData[callerID];
+  }
+}
+
+function setOpen(callerID) {
+  dataChannels[callerID].status = 'open';
+}
+
+function isOpen(callerID) {
+  if (dataChannels[callerID]) {
+    return dataChannels[callerID].status === 'open';
+  }
+
+  return false;
+}
+
+function isPending(callerID) {
```

Comment:
function isPending returns false if dataChannels[callerID] === undefined, which happens when any user leaves a room. If the same user rejoins the room they are assigned a different callerID - the IDs do not persist. I copied this function from the isOpen function on line 201 above, the functionality is identical.